src/main/java/sirius/db/jdbc/SQLQuery.java

```diff
@@ -127,39 +127,51 @@ public void iterate(Function<Row, Boolean> handler, @Nullable Limit limit) throw
                 if (stmt == null) {
                     return;
                 }
-                if (limit == null) {
-                    limit = Limit.UNLIMITED;
-                }
-                if (limit.getTotalItems() > 0) {
-                    stmt.setMaxRows(limit.getTotalItems());
-                }
-                if (limit.getTotalItems() > 1000 || limit.getTotalItems() <= 0) {
-                    if (ds.hasCapability(Capability.STREAMING)) {
-                        stmt.setFetchSize(Integer.MIN_VALUE);
-                    } else {
-                        stmt.setFetchSize(1000);
-                    }
-                }
+                Limit effectiveLimit = (limit == null) ? Limit.UNLIMITED : limit;
```

Comment:
I like it better with the default value at the end:
`(limit != null) ? limit : Limit.UNLIMITED;`